### PR TITLE
Refactor commands part3

### PR DESCRIPTION
**Pull request by bzeller**
The next set of commands migrated to the new API.

### CODE REVIEW

File: src/misc.h [hunk 1]
```diff
@@ -40,19 +45,19 @@ void remove_selections( Zypper & zypper );
...
  */
-void install_src_pkgs( Zypper & zypper );
+void install_src_pkgs( Zypper & zypper, zypp::DownloadMode dlMode_r );
 
 /**
  * Inject requirements of a source package or its build dependencies (depending
  * on command line options) to the pool.
  */
-void build_deps_install( Zypper & zypper );
+void build_deps_install(Zypper & zypper , const std::vector<std::__cxx11::string> &srcPkgs_r, bool buildDepsOnly_r);
```
Comment (mlandres) [inline src/misc.h]: zypper/src/misc.h:61:61: error: ‘__cxx11’ is not a member of ‘std’

File: src/commands/selectpatchoptionset.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  enum AnyTypeMode
+  {
+    EnableAnyType, //< Enables the arguments to take optional args and select any issue type if no arg was given,
+                   //  also enables the generic --issue flag
+    DisableAnyType
+  };
+
+  SelectPatchOptionSet ( ZypperBaseCommand &parent, AnyTypeMode mode_r = DisableAnyType );
+
+ using BaseCommandOptionSet::BaseCommandOptionSet;
```
Comment (mlandres) [inline src/commands/selectpatchoptionset.h]: Introduces `BaseCommandOptionSet::BaseCommandOptionSet(ZypperBaseCommand &parent )`.
Together with L25 `SelectPatchOptionSet ( ZypperBaseCommand &parent, AnyTypeMode mode_r = DisableAnyType )`:
```
In file included from /Local/ma/zypp/zypper/src/Command.cc:29:0:
/Local/ma/zypp/zypper/src/commands/patch.h:31:49: error: conversion from ‘<brace-enclosed initializer list>’ to ‘SelectPatchOptionSet’ is ambiguous
   SelectPatchOptionSet _selectPatchOpts { *this };
                                                 ^
/Local/ma/zypp/zypper/src/commands/patch.h:31:49: note: candidates are:
In file included from /Local/ma/zypp/zypper/src/commands/patch.h:12:0,
                 from /Local/ma/zypp/zypper/src/Command.cc:29:
/Local/ma/zypp/zypper/src/commands/selectpatchoptionset.h:27:30: note: SelectPatchOptionSet::SelectPatchOptionSet(ZypperBaseCommand&)
  using BaseCommandOptionSet::BaseCommandOptionSet;
                              ^
/Local/ma/zypp/zypper/src/commands/selectpatchoptionset.h:25:3: note: SelectPatchOptionSet::SelectPatchOptionSet(ZypperBaseCommand&, SelectPatchOptionSet::AnyTypeMode)
   SelectPatchOptionSet ( ZypperBaseCommand &parent, AnyTypeMode mode_r = DisableAnyType );
   ^
...
src/CMakeFiles/zypper_lib.dir/build.make:75: recipe for target 'src/CMakeFiles/zypper_lib.dir/Command.cc.o' failed
make[2]: *** [src/CMakeFiles/zypper_lib.dir/Command.cc.o] Error 1
```

Comment (mlandres) [reply src/commands/selectpatchoptionset.h]: ...and indent of the `using ...` line

File: src/misc.h (hunk 1 above)
Comment (bzeller) [reply src/misc.h]: Removed

File: src/commands/selectpatchoptionset.h (hunk 2 above)
Comment (bzeller) [reply src/commands/selectpatchoptionset.h]: Dropped the using

File: src/Summary.h [hunk 3]
```diff
@@ -46,6 +46,8 @@ class Summary : private base::NonCopyable
     SHOW_NOT_UPDATED        = 0x0800,
     SHOW_LOCKS              = 0x1000,
 
+    UPDATESTACK_ONLY        = 0x2000,  //< required for zypper patch
```
[the remark this answers is not in the document]
Comment (bzeller) [reply src/Summary.h]: As discussed, makes no difference in zypper since we do not build doxygen docs

**Pull request merged**